src/components/MultiHopTrade/components/TradeAssetInput.tsx

```diff
@@ -50,13 +52,16 @@ const AssetInputWithAsset: React.FC<AssetInputLoadedProps> = props => {
     if (props.onChange) props.onChange(balance, false)
   }, [balance, props])
 
-  const { assetNamespace } = fromAssetId(assetId)
+  const showMax = useMemo(
+    () => isEvmChainId(fromAssetId(assetId).chainId) && !isNativeEvmAsset(assetId),
+    [assetId],
+  )
```

Comment:
Golden.

Nitpick, re-name to `shouldShowMax` to disambiguate that this is not an action.